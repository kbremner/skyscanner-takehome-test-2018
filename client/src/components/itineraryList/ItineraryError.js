import React from 'react';
import RefreshIcon from 'bpk-component-icon/lg/refresh';
import { withButtonAlignment, withRtlSupport } from 'bpk-component-icon';
import BpkText from 'bpk-component-text';
import BpkButton from 'bpk-component-button';
import PropTypes from '../../types';
import './ItineraryError.scss';

const AlignedRefreshIcon = withButtonAlignment(withRtlSupport(RefreshIcon));

const ItineraryError = ({ className, onRetry }) => (
  <div className={[className, 'itinerary-error'].join(' ')}>
    <BpkText className="itinerary-error__text" textStyle="xl">Something's gone wrong!</BpkText>
    <BpkButton className="itinerary-error__retry-btn" secondary large onClick={onRetry}>
      <span className="itinerary-error__retry-text">Retry</span>
      <AlignedRefreshIcon />
    </BpkButton>
  </div>
);

ItineraryError.propTypes = {
  onRetry: PropTypes.func.isRequired,
  className: PropTypes.string
};

ItineraryError.defaultProps = {
  className: null
};

export default ItineraryError;
