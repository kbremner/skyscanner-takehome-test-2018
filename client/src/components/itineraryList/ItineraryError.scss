@import "~bpk-mixins";

.itinerary-error {
  display: flex;
  align-items: center;
  flex-direction: column;

  .itinerary-error__text {
    padding-top: $bpk-spacing-md;
  }
  
  .itinerary-error__retry-btn {
    margin: $bpk-spacing-md;
  }

  .itinerary-error__retry-text {
    padding-right: $bpk-spacing-sm;
  }
}