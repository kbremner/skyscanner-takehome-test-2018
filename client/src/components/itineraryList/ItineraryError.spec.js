import React from 'react';
import { shallow } from 'enzyme';
import ItineraryError from './ItineraryError';

const defaultProps = () => ({
  onRetry: jest.fn()
});

const setup = (overrides = {}) => {
  const props = {
    ...defaultProps(),
    ...overrides
  };

  const wrapper = shallow(<ItineraryError {...props} />);

  return { wrapper, props };
};

describe('ItineraryError', () => {
  it('renders', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('adds class name to card provided', () => {
    const { wrapper, props } = setup({ className: 'test-class-name' });
    expect(wrapper.hasClass(props.className)).toBe(true);
  });

  it('calls retry handler when button is clicked', () => {
    const { wrapper, props } = setup();
    wrapper.find('BpkButton').simulate('click');
    expect(props.onRetry).toHaveBeenCalledTimes(1);
  })
});
