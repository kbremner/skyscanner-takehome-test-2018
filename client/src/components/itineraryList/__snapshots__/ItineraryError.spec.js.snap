// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryError renders 1`] = `
<div
  className=" itinerary-error"
>
  <BpkText
    bold={false}
    className="itinerary-error__text"
    tagName="span"
    textStyle="xl"
  >
    Something's gone wrong!
  </BpkText>
  <BpkButton
    className="itinerary-error__retry-btn"
    destructive={false}
    disabled={false}
    featured={false}
    href={null}
    iconOnly={false}
    large={true}
    link={false}
    onClick={[MockFunction]}
    secondary={true}
    submit={false}
  >
    <span
      className="itinerary-error__retry-text"
    >
      Retry
    </span>
    <withAlignment(withRtlSupport(Component)) />
  </BpkButton>
</div>
`;
